hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/ReadableScheduleStoreTest.java

```diff
@@ -16,98 +16,127 @@
+    private Key schedulerKey = Key.newBuilder().ed25519(Bytes.fromHex(SCHEDULER_KEY_HEX)).build();
+    private AccountID payer = AccountID.newBuilder().accountNum(2001L).build();
+    private Key payerKey = Key.newBuilder().ed25519(Bytes.fromHex(PAYER_KEY_HEX)).build();
+    // spotless:on
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private Account schedulerAccount;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private Account payerAccount;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private ReadableAccountStore accountStore;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private ReadableStates states;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private ReadableKVStateBase<ScheduleID, Schedule> schedulesById;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private Schedule scheduleInState;
+
+    @Mock(strictness = Mock.Strictness.LENIENT)
+    private ReadableStoreFactory mockStoreFactory;
+
+    // Non-Mock objects, but may contain or reference mock objects.
+    private ReadableScheduleStore scheduleStore;
+    private SchedulableTransactionBody scheduled;
 
     @BeforeEach
-    void setUp() {
-        given(states.get("SCHEDULES_BY_ID")).willReturn(state);
-        subject = new ReadableScheduleStoreImpl(states);
+    void setUp() throws PreCheckException {
+        BDDMockito.given(states.<ScheduleID, Schedule>get("SCHEDULES_BY_ID")).willReturn(schedulesById);
+        BDDMockito.given(schedulerAccount.key()).willReturn(schedulerKey);
+        BDDMockito.given(payerAccount.key()).willReturn(payerKey);
+
+        scheduleStore = new ReadableScheduleStoreImpl(states);
+
+        BDDMockito.given(accountStore.getAccountById(scheduler)).willReturn(schedulerAccount);
+        BDDMockito.given(accountStore.getAccountById(payer)).willReturn(payerAccount);
+
+        BDDMockito.given(scheduleInState.hasPayerAccount()).willReturn(Boolean.TRUE);
+        BDDMockito.given(scheduleInState.payerAccount()).willReturn(payer);
+        BDDMockito.given(scheduleInState.schedulerAccount()).willReturn(adminAccount);
+        BDDMockito.given(scheduleInState.hasAdminKey()).willReturn(true);
+        BDDMockito.given(scheduleInState.adminKey()).willReturn(adminKey);
+        BDDMockito.given(scheduleInState.scheduledTransaction()).willReturn(scheduled);
+
+        BDDMockito.given(schedulesById.get(testScheduleID)).willReturn(scheduleInState);
+        BDDMockito.given(mockStoreFactory.getStore(ReadableScheduleStore.class)).willReturn(scheduleStore);
+        BDDMockito.given(mockStoreFactory.getStore(ReadableAccountStore.class)).willReturn(accountStore);
     }
 
     @Test
+    @SuppressWarnings("DataFlowIssue")
     void constructorThrowsIfStatesIsNull() {
-        assertThrows(NullPointerException.class, () -> new ReadableScheduleStoreImpl(null));
+        Assertions.assertThrows(NullPointerException.class, () -> new ReadableScheduleStoreImpl(null));
```

Comment:
General comment about the test files in this PR, we've officially decided to use Assert4j instead of the jupiter API

Reply:
Ah right, I seem to remember you mentioning something along those lines when you put up the PR. Sounds good!